sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -327,6 +327,18 @@ export interface ListRulesResponse extends Array<Rule> {
  * All operations return promises that resolve to an object that has the relevant output.
  * These objects also have a property called `_response` that you can use if you want to
  * access the direct response from the service.
+ *
+ * @throws RestError for all of the operations with code as -
+ * `UnauthorizedRequestError` when given request fails due to authorization problems,
+ * `MessageEntityNotFoundError` when requested messaging entity does not exist,
+ * `MessageEntityAlreadyExistsError` when requested messaging entity already exists,
+ * `InvalidOperationError` when requested operation is invalid and we encounter a 403 HTTP status code,
+ * `QuotaExceededError` when requested operation fails due to quote limits exceeding from service side,
+ * `ServerBusyError` when the request fails due to server being busy,
+ * `ServiceError` when receiving unrecognized HTTP status or for a scenarios such as
+ * bad requests or requests resulting in conflict operations,
+ * or the code can be a value from the standard set of HTTP status codes as documented at
+ * https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netframework-4.8
```

Comment:
Some of the above error codes do not apply to some of the methods on the client.
For example, 
- `MessageEntityNotFoundError` would not apply to create or list methods.
- `MessageEntityAlreadyExistsError` would not apply to delete methods.

I would suggest moving this information on error codes to docs on the individual methods

Reply:
It's not certain that the API can return only certain specific HTTP response codes right?
Is there any documentation or code that talks about above?

Reply:
We can follow the docs that the .Net SDK uses if needed.
But I based my above observations on the usage patterns and general expectations of a create, list and delete method.

Please follow up with @bterlson offline on this. Any agreement reached there is fine by me.

Reply:
I think I agree with @ramya-rao-a, since some of the errors aren't applicable all the time, we should move the error docs into the leaves so users aren't misled about what might come out of calling the method.

Reply:
Also, I think this is a better link re: HTTP status codes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status. .NET docs are a little too non-authoritative for me.

Reply:
Updated to move docs to individual operations as discussed